### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by cbachhuber**
Addressing #31:
- Link dogm privately against glew and related libraries
- Re-create deleted build folder (only during compilation in CI), as suggested by @TheCodez 
- Tidy up CI instructions (this is according to my taste, feel free to suggest changes)

### CODE REVIEW

**Comment by TheCodez** [comment]
This introduced some small CMake errors on windows. Will report back once I fixed them.

**Comment by TheCodez** [comment]
Could you please change line 20 in CMakelists.txt to
```
if (NOT WIN32)
	find_package(GLEW)
endif()
```

I will try to automate the dependency searching on Windows in the future.

**Comment by TheCodez** [comment]
Now it can't find `<glm/vec4.hpp>` when compiling, nice :grinning:

File: dogm/CMakeLists.txt [hunk 1]
```diff
@@ -84,6 +76,14 @@ add_library(dogm STATIC
 	${SRC_FILES}
 )
 
+target_include_directories(dogm
```
Comment (TheCodez) [inline dogm/CMakeLists.txt:79]: Changing this to `include_directories` fixes the glm problem.

**Comment by cbachhuber** [comment]
I implemented both your suggestions. Does that fix the Windows build?

File: .github/workflows/cmake_build.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name: C/C++ CI
 
-on: [push]
+on: [push, pull_request]
```
Comment (TheCodez) [inline .github/workflows/cmake_build.yml:3]: Nice 👍

**Comment by TheCodez** [comment]
Yes, thank you!

**Pull request merged**